### DISCUSSION TITLE
Run Nextcloud with a self-signed certificate if it is behind a reverse proxy

### DISCUSSION THREAD

**Discussion by Zoey2936** [nextcloud/all-in-one#684] [Ideas] [answered]
Hello,
it would be nice if it would be possible to run Nextcloud with a self-signed certificate if it is behind a reverse proxy, like the aio dashboard / mastercontainer. I know this would break all existing reverse proxy configurations, but if someone proxies not over localhost to the nextcloud apache container, the data will be sent over http and with that unencrypted through the internet. (for example, if the proxy runs on another server or the proxy runs inside a docker container without network mode host). Maybe there could be an option for this, or is this already possible?
Thanks!

**Comment by Zoey2936** [comment]
Any opinions about this? (because there are no reactions to this)

**Comment by szaimen** [comment, marked as answer]
I know it is not a perfect solution but what about running an additional reverse proxy with self-signed certificates on the same machine?

**Reply by Zoey2936** [reply]
Would be an option, but a very complicated one
